blocks/bricks/__init__.py

```diff
@@ -444,6 +444,38 @@ class Softmax(Activation):
     def apply(self, input_):
         return tensor.nnet.softmax(input_)
 
+    @application
+    def categorical_cross_entropy(self, y, x):
+        """Return computationally stable softmax cost.
+
+        Parameters
+        ----------
+        y : :class:`~tensor.TensorVariable`
+            In the case of a matrix argument, each slice along
+            axis represents one distribution. In the vector case, each
+            element represents the position of the '1' in a one hot-vector.
+        x : :class:`~tensor.TensorVariable`
+            Each slice along axis represents one distribution.
```

Comment:
Hmm, but `x` is not distribution but rather the energies that have to be transformed into a distribution!